marketplace-service/src/main/java/com/axonivy/market/github/service/impl/GHAxonIvyProductRepoServiceImpl.java

```diff
@@ -1,127 +1,284 @@
+      String tag) {
+    ProductModuleContent productModuleContent = new ProductModuleContent();
+    try {
+      List<GHContent> contents = getProductFolderContents(product, ghRepository, tag);
+      productModuleContent.setTag(tag);
+      getDependencyContentsFromProductJson(productModuleContent, contents);
+      GHContent readmeFile = contents.stream().filter(GHContent::isFile)
+          .filter(content -> ReadmeConstants.README_FILE.equals(content.getName())).findFirst().orElse(null);
+      if (Objects.nonNull(readmeFile)) {
+        String readmeContents = new String(readmeFile.read().readAllBytes());
+        if (hasImageDirectives(readmeContents)) {
+          readmeContents = updateImagesWithDownloadUrl(product, contents, readmeContents);
+        }
+        getExtractedPartsOfReadme(productModuleContent, readmeContents);
+      }
+    } catch (Exception e) {
+      log.error("Cannot get product.json and README file's content {}", e);
+      return null;
+    }
+    return productModuleContent;
+  }
+
+  private void getDependencyContentsFromProductJson(ProductModuleContent productModuleContent, List<GHContent> contents)
+      throws IOException {
+    GHContent productJsonFile = getProductJsonFile(contents);
+    if (Objects.nonNull(productJsonFile)) {
+      List<MavenArtifact> artifacts = convertProductJsonToMavenProductInfo(productJsonFile);
+      MavenArtifact artifact = artifacts.stream().filter(MavenArtifact::getIsDependency).findFirst().orElse(null);
+
+      if (Objects.nonNull(artifact)) {
+        productModuleContent.setIsDependency(Boolean.TRUE);
+        productModuleContent.setGroupId(artifact.getGroupId());
+        productModuleContent.setArtifactId(artifact.getArtifactId());
+        productModuleContent.setType(artifact.getType());
+        productModuleContent.setName(artifact.getName());
+      }
+    }
+  }
+
+  private static GHContent getProductJsonFile(List<GHContent> contents) {
+    return contents.stream().filter(GHContent::isFile)
+        .filter(content -> ProductJsonConstants.PRODUCT_JSON_FILE.equals(content.getName())).findFirst().orElse(null);
+  }
+
+  public String updateImagesWithDownloadUrl(Product product, List<GHContent> contents, String readmeContents)
+      throws IOException {
+    Map<String, String> imageUrls = new HashMap<>();
+    List<GHContent> productImages = contents.stream().filter(GHContent::isFile)
+        .filter(content -> content.getName().toLowerCase().matches(IMAGE_EXTENSION)).toList();
+    if (!CollectionUtils.isEmpty(productImages)) {
+      for (GHContent productImage : productImages) {
+        imageUrls.put(productImage.getName(), productImage.getDownloadUrl());
+      }
+    } else {
+      getImagesFromImageFolder(product, contents, imageUrls);
+    }
+    for (Map.Entry<String, String> entry : imageUrls.entrySet()) {
+      String imageUrlPattern = "\\(([^)]*?" + Pattern.quote(entry.getKey()) + "[^)]*?)\\)";
```

Comment:
how if we use:
private static final String REGEX = "\\(([^)]*?%s[^)]*?)\\)";
 String.format(REGEX , Pattern.quote(entry.getKey()))